src/ripple/core/impl/Workers.cpp

```diff
@@ -17,264 +17,119 @@
 */
 //==============================================================================
 
-#include <ripple/basics/PerfLog.h>
 #include <ripple/beast/core/CurrentThreadName.h>
 #include <ripple/core/impl/Workers.h>
 #include <cassert>
+#include <chrono>
+#include <thread>
 
 namespace ripple {
 
 Workers::Workers(
     Callback& callback,
-    perf::PerfLog* perfLog,
-    std::string const& threadNames,
-    int numberOfThreads)
-    : m_callback(callback)
-    , perfLog_(perfLog)
-    , m_threadNames(threadNames)
-    , m_allPaused(true)
-    , m_semaphore(0)
-    , m_numberOfThreads(0)
-    , m_activeCount(0)
-    , m_pauseCount(0)
-    , m_runningTaskCount(0)
+    std::string const& name,
+    unsigned int count)
+    : callback_(callback)
 {
-    setNumberOfThreads(numberOfThreads);
-}
-
-Workers::~Workers()
-{
-    stop();
-
-    deleteWorkers(m_everyone);
-}
-
-int
-Workers::getNumberOfThreads() const noexcept
-{
-    return m_numberOfThreads;
-}
+    assert(count != 0);
 
-// VFALCO NOTE if this function is called quickly to reduce then
-//             increase the number of threads, it could result in
-//             more paused threads being created than expected.
-//
-void
-Workers::setNumberOfThreads(int numberOfThreads)
-{
-    static int instance{0};
-    if (m_numberOfThreads == numberOfThreads)
-        return;
-
-    if (perfLog_)
-        perfLog_->resizeJobs(numberOfThreads);
+    // It is important that head and tail are equal but non-zero on
+    // startup: we use zero as a special value to indicate that the
+    // thread must terminate; if zero was a 'legal' value then we'd
+    // encounter the ABA problem with std::atomic::wait.
+    assert(head_ == tail_ && head_ != 0);
 
-    if (numberOfThreads > m_numberOfThreads)
+    while (count--)
     {
-        // Increasing the number of working threads
-        int const amount = numberOfThreads - m_numberOfThreads;
+        // We need to increment this outside of the thread, to avoid a
+        // subtle (if unlikely) race condition during shutdown.
+        ++threads_;
 
-        for (int i = 0; i < amount; ++i)
-        {
-            // See if we can reuse a paused worker
-            Worker* worker = m_paused.pop_front();
+        std::thread th(
+            [this, name](unsigned int instance) {
```

Comment:
Just a preference, but why capture `this` and `name` but not `instance` / `count`? Could pass them all as parameters and turn this lambda into a function pointer. No biggie, just wondering how the distinction was made.